pkg/log/entry.go

```diff
@@ -16,46 +18,52 @@ type Entry interface {
 	Infof(format string, args ...any)
 	Warnf(format string, args ...any)
 	Errorf(format string, args ...any)
+	Fatalf(format string, args ...any)
 
 	// Attach contextual information to the logging entry
 	WithFields(fields map[string]any) Entry
 
+	WithField(key string, value any) Entry
+
+	WithError(err error) Entry
+
+	WithContext(ctx context.Context) Entry
+
 	// SystemErr is a method that disects the error
 	// and logs the appropriate level and fields for it.
 	SystemErr(err error)
 }
 
 type entry struct {
-	*logrus.Entry
+	*slog.Logger
 }
 
 func (e *entry) WithFields(fields map[string]any) Entry {
-	ent := e.Entry.WithFields(fields)
-	return &entry{ent}
+	ent := e.WithFields(fields)
+	return ent
 }
```

Comment:
Isn't this straight up an infinite loop?

Reply:
will take care of it! I am noob😅